Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM ubuntu:18.04 AS build
+
+COPY . /src
+
+WORKDIR /src
+
+ENV CC gcc
+ENV CONTAINER ubuntu:18.04
+ENV BUILDTYPE debug
+ENV RUSTPROFILE minimal
+
+RUN ci/container_scripts/install_deps.sh
+
+RUN ci/container_scripts/install_extra_deps.sh
+
+ENV PATH "/root/.cargo/bin:${PATH}"
+RUN ci/container_scripts/build_and_install.sh
+
+CMD /root/.shadow/bin/shadow --version
```

Comment:
We can do it but we will not be able to run test in the same way due to the ENTRYPOINT. 
```
ENTRYPOINT ["/root/.shadow/bin/shadow"]
CMD ["--version"]
```

I think we can also do a symlink on the `/root/.shadow/bin/shadow`.

Reply:
Ah good point, I forgot about that. I think it would be okay to use the shadow entrypoint in the Dockerfile, and then when running the tests, add the flag `--entrypoint "/bin/bash -c"` in the `docker run` call.

But maybe leaving the default entrypoint is better since the user may also want to run the utility scripts. Rather than adding a symlink, I think it would be best to update the PATH for the root user to include `/root/.shadow/bin`.

Reply:
But without the CMD option I can't run `Shadow` with a conf file :/

Reply:
better ? :)

Reply:
Almost :) So the `ENTRYPOINT` needs to use the exec form that you used in your earlier comment, so the only change that needs to be made is from:

```
ENTRYPOINT shadow
```

to:

```
ENTRYPOINT ["/root/.shadow/bin/shadow"]
```

This makes the `docker run` command nicer, but I'll leave another comment about this below.

Reply:
Better, but one more small change :) The `ENTRYPOINT` needs to use the exec form that you used in your earlier comment, so the only change that needs to be made is from:

```
ENTRYPOINT shadow
```

to:

```
ENTRYPOINT ["shadow"]
```

This makes the docker run command nicer, but I'll leave another comment about this below.